jenkins/docker-ecr/docker-build-ecr.jenkinsfile

```diff
@@ -0,0 +1,145 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 3, unit: 'HOURS')
+    }
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+            args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+        }
+    }
+    parameters {
+        string(
+                defaultValue: 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>',
+                name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS',
+                description: 'The script path of the docker build script',
+                trim: true
+        )
+        string(
+                name: 'IMAGE_IN_STAGING',
+                description: 'Image name on stating. Eg: ci-runner:1.3.0',
+                trim: true
+        )
+        choice(
+                choices: ['staging', 'production'],
+                name: 'ENVIRONMENT',
+                description: 'Are we pushing docker images to staging (opensearchstaging) or production (opensearchproject) account.'
+        )
+        string(
+                name: 'IMAGE_IN_PROD',
+                description: 'Only required when ENVIRONMENT is Production - Image in prod (Destination to copy the image to). Eg: ci-runner:1.3.0',
+                trim: true
+        )
+    }
+    stages {
+        stage('check-params') {
+            steps {
+                script {
+                    if (ENVIRONMENT == 'production'
+                            && (!IMAGE_IN_STAGING.contains(":") || !IMAGE_IN_PROD.contains(":"))) {
+                        currentBuild.result = 'ABORTED'
+                        error('IMAGE_IN_STAGING parameter or IMAGE_IN_PROD is incorrect')
+                    }
+                    if(ENVIRONMENT == 'staging'
+                            && (DOCKER_BUILD_SCRIPT_WITH_COMMANDS == null
+                            || DOCKER_BUILD_SCRIPT_WITH_COMMANDS == 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>')) {
```

Comment:
Since this value is reused, lets put it in a variable so it isn't repeated

Reply:
The command helps run the build script for docker from any repository. Changed the params and jobs now